staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go

```diff
@@ -46,7 +46,7 @@ func PeriodicallyInvalidate(cache AvailableAPIsGetter, interval time.Duration, s
 		case <-ticker.C:
 			cache.Invalidate()
 		case <-stopCh:
-			break
+			return
```

Comment:
This changes behavior, is there no affected test? :/ (to be clear, the new behavior of actually exiting looks better)

cc @logicalhan

Reply:
lgtm but yeah maybe we should add a test here.